SmokeSheep/SmokeSheep/class/SceneManager/Scene/TitleScene/TitileObj/TitleAnimations/TitleAnimations.cpp

```diff
@@ -0,0 +1,77 @@
+﻿#include "TitleAnimations.h"
+
+void TitleAnimations::Init()
+{
+	m_rGameLib.CreateTex(m_filePathAndKeys[TA_SHEEP][0], m_filePathAndKeys[TA_SHEEP][1]);
+	m_rGameLib.CreateTex(m_filePathAndKeys[TA_GRANDMA][0], m_filePathAndKeys[TA_GRANDMA][1]);
+}
+
+void TitleAnimations::Render()
+{
+	SheepAnimation();
+	GrandmaAnimation();
+}
+
+void TitleAnimations::SheepAnimation()
+{
+	CustomVertex sheep[4];
+
+	m_animationDatas[TA_SHEEP].m_center = { m_WND_SIZE.m_x * 0.15f,m_WND_SIZE.m_y * 0.9f, 0.0f };
+	m_animationDatas[TA_SHEEP].m_halfScale = { m_WND_SIZE.m_x * 0.05f, m_WND_SIZE.m_y * 0.25f , 0.0f };
+
+	float& startTU = m_animationDatas[TA_SHEEP].m_texUV.m_startTU;
+	float& endTU = m_animationDatas[TA_SHEEP].m_texUV.m_endTU;
+	
+	const int NUMBER_OF_IMAGES = 4;			//画像の切り取り数
+	const float IMAGE_SIZE = 250.f;			//統合ファイルの1枚の画像のサイズ
+	const float MAX_IMAGE_SIZE = 2048.f;	//画像の大きさ
+	
+	static int nextImageCount = 0;	//次のアニメーション画像に行くためのフレームカウント
+	nextImageCount += 1;
+
+	if (nextImageCount == 13)
+	{
+		m_animationCounts[TA_SHEEP] = (m_animationCounts[TA_SHEEP] == NUMBER_OF_IMAGES - 1) ? 0 : ++m_animationCounts[TA_SHEEP];
+		nextImageCount = 0;
+	}
+
+	startTU = (IMAGE_SIZE * m_animationCounts[TA_SHEEP]) / MAX_IMAGE_SIZE;
```

Comment:
```suggestion

const float 一枚目のUVの幅 = (IMAGE_SIZE / MAX_IMAGE_SIZE) 
startTU = 一枚目のUVの幅* m_animationCounts[TA_SHEEP];
```
上記のようにするといちいち計算処理しなくなるので個人的には好きな方法です。

Reply:
その方法にします
ありがとうございます